### DISCUSSION TITLE
How To Send Raw Request?

### DISCUSSION THREAD

**Discussion by Phoenix1112** [projectdiscovery/httpx#511] [Q&A]
hello. thanks for this project. i want to send raw request. But i did not do it. 

example request(request.txt):

```
GET /test HTTP/1.1
Host: localhost
User-Agent: Mozilla/5.0 (iPhone; CPU iPhone OS 6_0 like Mac OS X) AppleWebKit/536.26 (KHTML, like Gecko) Version/6.0 Mobile/10A5376e Safari/8536.25
Accept-Encoding: gzip, deflate
Connection: close
```

i am using this command. but no request.

```
httpx -request request.txt
```

Doesn't the httpx project have such a feature? Or am I using the wrong command?

**Comment by shubhack-repo** [comment]
Need raw request to perform some post sqli fuzzing or soap endpoints fuzzing?

**Comment by shenliehuozhi** [comment]
dou you know how to send raw request? if you know, can you tell me how please。

**Reply by olearycrew** [reply]
You can use -request with a file that contains the raw request

**Comment by D0nk3rS100t** [comment]
Still seeing this issue when using the -request option with a file containing a raw http request.

Version: 1.6.9 

Not seeing any errors or other output. Nothing showing up in upstream proxy logs either (assuming the request is simply not sent).
![image](https://github.com/user-attachments/assets/eaf47243-7742-4e77-a657-13d03ace2d37)
![image](https://github.com/user-attachments/assets/59e0acd7-4a0b-4925-8af5-3d0a802d09d3)

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
